core/mule.c

```diff
@@ -32,6 +32,24 @@ int mule_send_msg(int fd, char *message, size_t len) {
 	return 0;
 }
 
+// Runs in a mule thread and periodically scans for memory usage.
+static void *mule_mem_collector(void *foobar) {
+	// block all signals
+        sigset_t smask;
+        sigfillset(&smask);
+        pthread_sigmask(SIG_BLOCK, &smask, NULL);
+	int id = uwsgi.muleid;
+	uwsgi_log_verbose("mem-collector thread started for mule %d\n", id);
+	for(;;) {
+		sleep(uwsgi.mem_collector_freq);
+		uint64_t rss = 0, vsz = 0;
```

Comment:
this looks like a fix we should backport to the worker mem collector as on linux at least if there's not /proc/self/stat we are dereferencng that pointer

Reply:
Hello @xrmx ,

thanks for checking in. I've fixed the spaces and another fix related to the default frequency.

I'm not exactly clear what issue is solved that you'd want to backport - I have literally (including spaces) copied the mem collector from `core/uwsgi.c:3424`, but decided to always run it in a mule, not just when `-evil-reload-on-{as,rss}`.

Mind pointing me in the right direction?

Thanks,
   Arne

Reply:
See get_memusage:

```
        procfile = fopen("/proc/self/stat", "r");
        if (procfile) {
                i = fscanf(procfile, "%*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %*s %llu %lld", (unsigned long long *) vsz, (unsigned long long *) rss);
                ...
        }
        *rss = *rss * uwsgi.page_size;
```
And mem_collector:

```
                uint64_t rss, vsz;
                get_memusage(&rss, &vsz);
```

If we cannot read procfile rss is garbage

Reply:
Actually i was looking at uwsgi-2.0 branch which missed some backports for some stuff already fixed in master :)